lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -460,6 +480,28 @@ def build_merchant_data(xml, data, options = {})
         end
       end
 
+      def build_merchant_mpi_external(xml, data)
+        return unless data[:txid]
```

Comment:
The `txid` is only a required field for 3DS1, not 3DS2. Maybe something like:

```suggestion
        return unless data[:txid] || data[:threeDSServerTransID]
```